### PR TITLE
Do not enumerate all metadata objects when catalog is wrong

### PR DESCRIPTION
**Pull request by kokosing**
Do enumerate all metadata objects when catalog is wrong

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaMetadata.java [hunk 1]
```diff
@@ -291,6 +291,11 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
             return ImmutableSet.of();
         }
 
+        Optional<Set<String>> catalogs = filterString(constraint.getSummary(), CATALOG_COLUMN_HANDLE);
```
Comment (sopel39) [inline presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaMetadata.java]: How can it be wrong?

Comment (kokosing) [reply presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaMetadata.java]: yes it is, so when user specifies something wrong like `WHERE table_catalog = 'out_of_nowhere'` then there is no point to enumerate metadata objects.

Comment (kokosing) [reply presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaMetadata.java]: Please see added test

**Comment by findepi** [comment]
> Do enumerate all metadata objects when catalog is wrong

@kokosing 
`Do not enumerate all metadata objects when catalog is wrong` or
`Enumerate all metadata objects when catalog is wrong` ?

File: presto-tests/src/test/java/io/prestosql/tests/TestInformationSchemaConnector.java [hunk 2]
```diff
@@ -114,6 +114,7 @@ public void testLargeData()
         assertQuery("SELECT count(*) from test_catalog.information_schema.tables WHERE table_name LIKE 'test_t_ble1'", "VALUES 2");
         assertQuery("SELECT count(*) from test_catalog.information_schema.tables WHERE table_name LIKE 'test_t_ble1' AND table_name IN ('test_table1', 'test_table2')", "VALUES 2");
         assertQuery("SELECT count(*) from test_catalog.information_schema.columns WHERE table_schema = 'test_schema1' AND table_name = 'test_table1'", "VALUES 100");
+        assertQuery("SELECT count(*) from test_catalog.information_schema.columns WHERE table_catalog = 'wrong'", "VALUES 0");
```
Comment (findepi) [inline presto-tests/src/test/java/io/prestosql/tests/TestInformationSchemaConnector.java]: Is this correctness fix or speedup (short-circuiting)?

**Comment by kokosing** [comment]
@findepi 

Sorry about this should be: `Do not enumerate all metadata objects when catalog is wrong`

File: presto-tests/src/test/java/io/prestosql/tests/TestInformationSchemaConnector.java (hunk 2 above)
Comment (kokosing) [reply presto-tests/src/test/java/io/prestosql/tests/TestInformationSchemaConnector.java]: speedup (short-circuiting)

Comment (findepi) [reply presto-tests/src/test/java/io/prestosql/tests/TestInformationSchemaConnector.java]: (in the latter case, it'd be good to have a test that counts metadata calls)

Comment (kokosing) [reply presto-tests/src/test/java/io/prestosql/tests/TestInformationSchemaConnector.java]: > (in the latter case, it'd be good to have a test that counts metadata calls)

There is such, notice that count of metadata calls didn't change.

**Comment by kokosing** [comment]
@findepi comments addressed

File: presto-tests/src/test/java/io/prestosql/tests/TestInformationSchemaConnector.java [hunk 3]
```diff
@@ -175,4 +220,88 @@ private static DistributedQueryRunner createQueryRunner()
...
+        }
+
+        private MetadataCallsCount(long listSchemasCount, long listTablesCount, long getColumnsCount)
+        {
+            this.listSchemasCount = listSchemasCount;
+            this.listTablesCount = listTablesCount;
+            this.getColumnsCount = getColumnsCount;
+        }
+
+        public MetadataCallsCount setListSchemasCount(long listSchemasCount)
```
Comment (findepi) [inline presto-tests/src/test/java/io/prestosql/tests/TestInformationSchemaConnector.java]: `withListSchemasCount`

**Pull request merged**